### PR TITLE
*: update the algorithm for bounding regions

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Factor this to keep track of two points, instead of doing floating-point arithemtic, as that is lossy over many iterations and impossible to test. Fixed a small bug in the latitude delta during this refactor.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: components/helpers/geographicCoordinates.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      bounds.bottomRight.longitude = coordinate.longitude;
+    }
+
+    if (coordinate.latitude > bounds.topLeft.latitude) {
+      bounds.topLeft.latitude = coordinate.latitude;
+    }
+    if (coordinate.latitude < bounds.bottomRight.latitude) {
+      bounds.bottomRight.latitude = coordinate.latitude;
+    }
+  }
```
Comment (floatplane) [inline components/helpers/geographicCoordinates.ts:49]: I think these conditionals get simpler with min/max

```suggestion
  for (const coordinate of coordinates) {
    // initialize our points to something on the polygons, so we always
    // end up centered around the polygons we're bounding
    if (bounds.topLeft.longitude === 0) {
      bounds.topLeft.longitude = coordinate.longitude;
      bounds.topLeft.latitude = coordinate.latitude;
      bounds.bottomRight.longitude = coordinate.longitude;
      bounds.bottomRight.latitude = coordinate.latitude;
    } else {
      bounds.topLeft.longitude = Math.min(coordinate.longitude, bounds.topLeft.longitude);
      bounds.bottomRight.longitude = Math.max(coordinate.longitude, bounds.bottomRight.longitude);
      bounds.topLeft.latitude = Math.max(coordinate.longitude, bounds.topLeft.longitude);
      bounds.bottomRight.latitude = Math.min(coordinate.longitude, bounds.bottomRight.longitude);
    }
  }
```

File: components/helpers/geographicCoordinates.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      topLeft: {latitude: 45.4921, longitude: -121.952},
+      bottomRight: {latitude: 45.1916, longitude: -121.431},
+    };
+
+    it('minimally bounds all coordinates with the region', () => {
+      const bounds: RegionBounds = {
+        topLeft: {latitude: 0, longitude: 0},
+        bottomRight: {latitude: 0, longitude: 0},
+      };
+      expect(updateBoundsToContain(bounds, coordinates)).toStrictEqual(regionBounds);
```
Comment (floatplane) [inline components/helpers/geographicCoordinates.test.ts:69]: for future reference: you might be interested in https://github.com/maasencioh/jest-matcher-deep-close-to

**Review by floatplane — COMMENTED** [review]
one suggestion!

**Comment by stevekuznetsov** [comment]
@floatplane thanks - that's way cleaner now :)

**Review by floatplane — APPROVED** [review]

**Pull request merged**